### PR TITLE
HUE-9219 [parser] Improve PartitionBy rules in impala

### PR DESCRIPTION
**Pull request by HSunboy**
PARTITIONED BY (col_name[, ...]) rule is missing in the impala grammer file.
image

### CODE REVIEW

**Comment by romainr** [comment]
Nice, thanks! BTW, have you looked at adding tests? e.g. https://docs.gethue.com/developer/parsers/#extending-the-grammar It saves a lot of time on the long run and even when developing in TDD.

**Comment by HSunboy** [comment]
> Nice, thanks! BTW, have you looked at adding tests? e.g. https://docs.gethue.com/developer/parsers/#extending-the-grammar It saves a lot of time on the long run and even when developing in TDD.

Good advice! I have found some wrong test cases and fixed them.

File: desktop/core/src/desktop/js/parse/jison/sql/impala/sql_create.jison [hunk 1]
```diff
@@ -560,7 +560,13 @@ PartitionedBy
  | 'PARTITION' 'BY' HashClauses
  | 'PARTITION' 'BY' HashClauses ',' RangeClause
  | 'PARTITION' 'BY' RangeClause
+ | 'PARTITIONED' 'BY' '(' ColumnIdentifierList ')'
```
[the remark this answers is not in the document]
Comment (HSunboy) [reply desktop/core/src/desktop/js/parse/jison/sql/impala/sql_create.jison:563]: ParenthesizedColumnSpecificationList consists of col_name, type and comment.
ColumnIdentifierList is only consists of col_name.
'Create...PartitionBy...As Select...' only need col_name.

[the remark this answers is not in the document]
Comment (HSunboy) [reply desktop/core/src/desktop/js/parse/jison/sql/impala/sql_create.jison:563]: See this:
image

**Comment by romainr** [comment]
Nice, indeed looks like it is column name name only for CTAS https://docs.cloudera.com/documentation/enterprise/6/6.3/topics/impala_create_table.html

File: desktop/core/src/desktop/js/parse/jison/sql/impala/sql_create.jison (hunk 1 above)
[the remark this answers is not in the document]
Comment (JohanAhlen) [reply desktop/core/src/desktop/js/parse/jison/sql/impala/sql_create.jison:563]: Ah yes, sorry, I see the difference. Thanks.

**Pull request merged**